### DISCUSSION TITLE
Simple Bottom Tab implementation help

### DISCUSSION THREAD

**Discussion by jsh088** [infinitered/ignite#1827] [Q&A]
I wanted to convert the navigation to Bottom Tabs. 
When I try it, my tabs and navigation loads, but the content is blank for all tabs...

The simple examples I am trying to implement works fine in code sandboxes online.

I have also tried using screens from the generator and such with same results.

```
app-navigator.tsx

function HomeScreen() {
  return (
    <View style={{ flex: 1, justifyContent: 'center', alignItems: 'center' }}>
      <Text>Home!</Text>
    </View>
  );
}

function SettingsScreen() {
  return (
    <View style={{ flex: 1, justifyContent: 'center', alignItems: 'center' }}>
      <Text>Settings!</Text>
    </View>
  );
}

const Tab = createBottomTabNavigator();
export const AppNavigator = () => {
  return (
    <NavigationContainer>
      <Tab.Navigator>
        <Tab.Screen name="Home" component={HomeScreen} />
        <Tab.Screen name="Sett" component={SettingsScreen} />
      </Tab.Navigator>
    </NavigationContainer>
  )
}
```

Any hints or suggestions on what I am missing? I am a bit lost as to what to try next.

**Comment by leonskim** [comment]
@jsh088 Hi there!

I've applied your code into the newly brewed Ignite app and tried running, it works fine on my end. (tested project: https://github.com/leonskim/bottom-tab) I assume there could be some configuration affecting your issue. Would you be able to share your project so I could take a look?

Thanks!

### Working example with the code provided
![111](https://user-images.githubusercontent.com/8325407/143152234-2f0ef3e6-128a-43de-9fcf-d1c13c30d029.gif)

**Reply by jsh088** [reply]
Hi, thanks so much for looking into this.
I tried it again on another native build and it seems to work as expected...
So I tried a few more things and figured out the problem occurs for projects created with the --expo option, probably related to https://github.com/infinitered/ignite/issues/1735

**Reply by leonskim** [reply]
@jsh088  I just created another project with `--expo` and used the same code as above, works the same on my end. Also I read your comment on #1735, getting the expo ready takes time but it shouldn't take too long. (Just an FYI, the entire directory including `node_modules` sizes 850MB on my just ignited Expo project)

Have you tried with the latest Ignite(master branch)? Expo and RN versions were updated in the master branch recently, the issue might be gone so please give it a try!

**Reply by jsh088** [reply]
hm... I've been trying a new project every other day or so. i'll give it another try.

the --expo option runs for more than an hour without finishing for me with constant disk usage on the Git for Windows process. I've never experienced any project take longer than 15 minutes (or even >5 minutes) on my hardware and connection so seems very highly suspect. I would normally download several tens of gigs within a few minutes time, so 1gig really is nothing. The install for a new ignite react native project takes ~3 minutes for me. 
If you see the other posts in there, i think lots of them mention running for hours without finishing as well. Seemed like it was in-line with what I am experiencing.
How long does it take you to run it?

**Reply by leonskim** [reply]
I haven't tried with the older versions but with the latest `master` branch on my laptop (2016 MBP 13"), it took less than 10 mins to initialize an Expo project. Have you been trying with the latest?

**Reply by jsh088** [reply]
uh.. does `npx ignite-cli new PizzaApp --expo` not use the `master`?

**Reply by leonskim** [reply]
Yes it does. I thought you've been trying with a old version of Ignite. (My bad!) 

In fact, there are a couple users reporting the same issue on our community Slack channel. To diagnose, would you mind posting the result of the command?

```
npx ignite-cli new PizzaApp --expo --verbose --debug
```

Also this might be helpful too:

```
npx ignite-cli doctor
```

**Reply by jsh088** [reply]
i think it gets stuck here:

` [object Object]


   · · · · · · · · · · · · · · · · · · 🔥 Ignite 🔥 · · · · · · · · · · · · · · · · · ·

    █ Creating PZAXP211130B using Ignite 7.8.0
    █ Powered by Infinite Red - https://infinite.red
    █ Using expo-cli
    █ Bundle identifier: com.PZAXP211130B
    ────────────────────────────────────────────────

 [object Object]
- Igniting app
 npx expo-cli@4 init PZAXP211130B --template xxx\AppData\Local\npm-cache\_npx\e31027f3785124a8\node_modules\ignite-cli\boilerplate --non-interactive
    🔥 Igniting app
 - Downloading and extracting project files.

 √ Downloaded and extracted project files.

 [19:50:37] 🧶 Using Yarn to install packages. Pass --npm to use npm instead.

- Summoning Expo CLI
 - Installing JavaScript dependencies.

 > yarn install

 - Installing JavaScript dependencies.

 yarn install v1.19.1

 info No lockfile found.

 [1/4] Resolving packages...

 [2/4] Fetching packages...

 info fsevents@2.3.2: The platform "win32" is incompatible with this module.

 info "fsevents@2.3.2" is an optional dependency and failed compatibility check. Excluding it from installation.

 info fsevents@1.2.13: The platform "win32" is incompatible with this module.

 info "fsevents@1.2.13" is an optional dependency and failed compatibility check. Excluding it from installation.

 [3/4] Linking dependencies...

 [4/4] Building fresh packages...

 success Saved lockfile.

 $ node ./bin/postInstall

 patch-package 6.2.2

 Applying patches...



 ✔︎ Solidarity checks valid

 Jetifier found 1249 file(s) to forward-jetify. Using 24 workers...

 Done in 94.41s.

 √ Installed JavaScript dependencies.

 - Installing CocoaPods...
√ Skipped installing CocoaPods because operating system is not on macOS.



 ✅ Your project is ready!

    🪔 Summoning Expo CLI
- Cleaning up Expo install

To run your project, navigate to the directory and run one of the following yarn commands.

- cd PZAXP211130B
- yarn start # you can open iOS, Android, or web from here, or run them directly with the commands below.
- yarn android
- yarn ios # requires an iOS device or macOS for access to an iOS simulator
- yarn web

`

I don't know if its related, but I also can't get the ignite non --expo project to run `yarn web` or `expo start --web` correctly after installing expo into the project for running on web... just plain expo projects work fine.

**Reply by leonskim** [reply]
Thanks for the log. I have 2 quick questions.

1. It seems like it was stuck while running `npx expo-cli init` command. I think something failed while it was running. What's your `expo-cli` version? (And could you try again after upgrading it if it's not the latest?)

2. I'm afraid I'm not quite understanding this. How did you install expo into the bare RN project created via Ignite (non --expo Ignite project)? 
> I also can't get the ignite non --expo project to run yarn web or expo start --web correctly after installing expo into the project for running on web

**Reply by jsh088** [reply]
1. 4.13.0
2. I thought you could install expo into an RN project but I guess not? It was just something I tried, wasn't sure if it works that way

**Reply by leonskim** [reply]
Gotcha. If you add `expo` to your bare RN project, it doesn't make the bare RN project to "expo-managed RN project". What it actually is a `Expo Modules`(It used to be `react-native-unimodules`). That allows you to use Expo modules like `expo-camera` in your bare RN apps. Unfortunately, since it doesn't make your app "expo-managed", you can't use `yarn web` or `expo start --web`.

As far as the issue goes, it doesn't seem like `expo-cli` causing it. I'm looking into `install-colo-loco`, which might affect the setting up process.

**Reply by leonskim** [reply]
There's an update that might fix the hang on initiating projects.(https://github.com/infinitered/ignite/pull/1842, merged to `master`) Give it a try and let me know if that works for you.

**Reply by jsh088** [reply]
still gets stuck in the same place after `Cleaning up Expo install`. log looks the same

**Reply by jamonholmgren** [reply]
FYI I am planning to remove colo-loco by default.

**Reply by jamonholmgren** [reply]
#1890 FYI
